ksp_plugin/vessel.cpp

```diff
@@ -598,15 +564,11 @@ Status Vessel::FlowPrognostication(
       << "Prognostication from " << prognosticator_parameters.first_time
       << " finished at " << prognostication->back().time << " with "
       << status.ToString() << " for " << ShortDebugString();
-  return status;
-}
-
-void Vessel::SwapPrognostication(
-    std::unique_ptr<DiscreteTrajectory<Barycentric>>& prognostication,
-    Status const& status) {
-  prognosticator_lock_.AssertHeld();
-  if (status.error() != Error::CANCELLED) {
-    prognostication_.swap(prognostication);
+  if (status.error() == Error::CANCELLED) {
+    return status;
+  } else {
+    // Ignore the status in this case and try to use the prognostication.
```

Comment:
`in this case` is not very helpful, this is `unless we were stopped`.
`try to use` is weird, this code is not the one using it, and the main thread may not do anything with it; `provide` the prognostication (implicitly to the main thread).

`Unless we were stopped, ignore the status, which indicates a failure to reach |t_max|, and provide a short prognostication.`